### DISCUSSION TITLE
Declare variables file and use in other files

### DISCUSSION THREAD

**Discussion by assayire** [hairyhenderson/gomplate#2226] [Q&A]
Hi,

How can I declare a `vars.tmpl` file with variables as a source of truth, and be able to include/use `vars.tmpl` in other files?

I tried a few different ways based on what I understood from the documentation, and also extensive web search. None of them worked ☹️ 

Is there a pattern available for this?

Thanks 🙏

**Comment by assayire** [comment]
@hairyhenderson Soft nudge 🙏

**Reply by tewfik-ghariani** [reply]
You may export environment variables and then access them with gomplate as follows

```
  custom_var: |-
    {{ .Env.CUSTOM_VAR }}
```

**Reply by hairyhenderson** [reply]
That's one way - thanks @tewfik-ghariani!

Another approach is to use nested templates. See https://docs.gomplate.ca/usage/#--template-t
